crates/sui-core/src/authority/test_authority_builder.rs

```diff
@@ -65,6 +83,11 @@ impl<'a> TestAuthorityBuilder<'a> {
         self
     }
 
+    pub fn with_keypair(mut self, keypair: &'a AuthorityKeyPair) -> Self {
```

Comment:
This API gets tricky because it conflicts with with_genesis_and_keypair. Is it definitely needed?

Reply:
It's not definitely needed. Do you think it's better to get rid of `with_genesis_and_keypair` and have separate `with_genesis`, `with_keypair` though?